src/core/load-fragment.c

```diff
@@ -5404,6 +5405,64 @@ int config_parse_ip_filter_bpf_progs(
         return 0;
 }
 
+int config_parse_pinned_bpf_progs(
+                const char *unit,
+                const char *filename,
+                unsigned line,
+                const char *section,
+                unsigned section_line,
+                const char *lvalue,
+                int ltype,
+                const char *rvalue,
+                void *data,
+                void *userdata) {
+        _cleanup_free_ char *resolved = NULL;
+        _cleanup_free_ char *bpffs_path = NULL;
+        enum bpf_attach_type attach_type;
+
+        const Unit *u = userdata;
+        CGroupContext *c = data;
+        int r;
+
+        assert(filename);
+        assert(lvalue);
+        assert(rvalue);
+        assert(c);
+
+        if (isempty(rvalue)) {
+                while (c->pinned_bpf_progs)
+                        cgroup_context_free_pinned_bpf_program(c, c->pinned_bpf_progs);
+                r = SYNTHETIC_ERRNO(ENOENT);
+                log_syntax(unit, LOG_WARNING, filename, line, r, "Unit specifiers are empty, cleaning BPF program context");
```

Comment:
does this warrant a LOG_WARNING? It's pretty much standard that empty line resets previous config